webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -57,31 +63,31 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
     }
 
     @Override
-    public List<CsvValidationFailure> validateHeaders(final List<String> csvHeaders) {
-        final List<CsvValidationFailure> failures = new ArrayList<>();
+    public List<CsvValidationResult> validateHeaders(final List<String> csvHeaders) {
+        final List<CsvValidationResult> results = new ArrayList<>();
 
         for (int i = 0; i < RequiredHeaders.size(); i++) {
             if (!RequiredHeaders.get(i).equalsIgnoreCase(csvHeaders.get(i))) {
-                failures.add(CsvValidationFailure.builder()
+                results.add(CsvValidationResult.builder()
                         .row(0)
                         .message("Invalid headers. Headers must be in order: [" +
                                 Joiner.on(",").join(RequiredHeaders) +
                                 "]")
+                        .isOk(false)
                         .build());
                 break;
             }
         }
-        return failures;
+        return results;
     }
 
     @Override
-    public List<CsvValidationFailure> validateRecords(final Iterator<CSVRecord> recordIterator,
-                                                      final PermissionScope permissionScope) {
-        final List<CsvValidationFailure> validationFailures = new ArrayList<>();
+    public List<CsvValidationResult> validateRecords(final Iterator<CSVRecord> recordIterator,
+                                                     final PermissionScope permissionScope) {
+        final List<CsvValidationResult> validationResults = new ArrayList<>();
 
-        final Set<String> availableSchools = schoolRepository.findAll(permissionScope).stream()
-                .map(Organization::getNaturalId)
-                .collect(Collectors.toSet());
+        final List<Organization> availableSchools = schoolRepository.findAll(permissionScope).stream()
```

Comment:
I feel like guava would have a one-liner for this that isn't as resource wasteful as using a stream

Reply:
I actually don't need this stream at all anymore since we are no longer mapping just the `naturalId`